src/material/dialog/dialog.ts

```diff
@@ -144,13 +148,23 @@ export abstract class _MatDialogBase<C extends _MatDialogContainerBase> implemen
       (typeof ngDevMode === 'undefined' || ngDevMode)) {
       throw Error(`Dialog with id "${config.id}" exists already. The dialog id must be unique.`);
     }
+    // If there are open dialogs, and the last opened dialog was of the same component or template
+    // ref as the one opening now, check if we're still within the animation transition time of the
+    // previous dialog, if so, just return the MatDialogRef of the previous dialog.
+    if (this.openDialogs.length && componentOrTemplateRef === this._lastDialogOpenType &&
+        this._dialogOpenedWithinPreviousTransitionTime()) {
+      return this._lastDialogRef;
+    }
 
     const overlayRef = this._createOverlay(config);
     const dialogContainer = this._attachDialogContainer(overlayRef, config);
     const dialogRef = this._attachDialogContent<T, R>(componentOrTemplateRef,
                                                       dialogContainer,
                                                       overlayRef,
                                                       config);
+    this._lastDialogOpenTime = new Date().getTime();
+    this._lastDialogOpenType = componentOrTemplateRef;
+    this._lastDialogRef = dialogRef;
```

Comment:
I wonder if these should get set to `null` in `ngOnDestroy()`? But I guess that would require changing the types to accept `null`.

Reply:
I don't get why we need to deal with time at all. What's stopping from setting an `isAnimating` flag on the dialog container and reading it instead? We already have all the event handlers in place and the dialog refs are already tracked in `MatDialog.openDialogs`.

Reply:
That's what caused the need for all of the flushing in the other PR. But I guess you feel like this specific approach would avoid that?

Reply:
I thought that the flushing was necessary, because we introduced a hard error if you didn't flush. Since now we return a dialog ref, the flushing should be necessary.

Reply:
Confirmed, when I removed the throwing of the error, I was able to remove the `flush();` calls from the other PR.